File: android/src/oldArch/java/com/spo/nfcscanner/nfc/NfcLifecycleGuardService.java
package com.spo.nfcscanner.nfc;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

/**
 * Lightweight service whose sole purpose is to observe task removal events so NFC dispatch
 * can be disabled before the process is terminated.
 */
public class NfcLifecycleGuardService extends Service {

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        // We do not need to restart automatically if the system kills the process.
        return START_NOT_STICKY;
    }

    @Override
    public void onTaskRemoved(Intent rootIntent) {
        super.onTaskRemoved(rootIntent);
        NfcController.getInstance().disableDispatchHandling();
        stopSelf();
    }

    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }
}

File: android/src/main/java/com/spo/nfcscanner/nfc/NfcLifecycleGuardService.kt
package com.spo.nfcscanner.nfc

import android.app.Service
import android.content.Intent
import android.os.IBinder

/**
 * Lightweight service whose sole purpose is to observe task removal events so NFC dispatch
 * can be disabled before the process is terminated.
 */
class NfcLifecycleGuardService : Service() {

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        // We do not need to restart automatically if the system kills the process.
        return START_NOT_STICKY
    }

    override fun onTaskRemoved(rootIntent: Intent?) {
        super.onTaskRemoved(rootIntent)
        NfcController.disableDispatchHandling()
        stopSelf()
    }

    override fun onBind(intent: Intent?): IBinder? = null
}
